repose-aggregator/extensions/simple-rbac/src/main/scala/org/openrepose/filters/simplerbac/SimpleRbacFilter.scala

```diff
@@ -0,0 +1,261 @@
+    }
+  }
+
+  override def isInitialized: Boolean = initialized
+
+  private def getHandlers: DispatchHandler = {
+    val handlers: util.List[ResultHandler] = new util.ArrayList[ResultHandler]
+    Option(configuration.getDelegating) match {
+      case Some(d) =>
+        handlers.add(new MethodLabelHandler)
+        handlers.add(new DelegationHandler(d.getQuality))
+      case _ =>
+        handlers.add(new ServletResultHandler)
+    }
+    handlers.add(new InstrumentedHandler)
+    handlers.add(new ApiCoverageHandler)
+    Option(configuration.getDotOutput) match {
+      case Some(d) =>
+        if (StringUtils.isNotBlank(d)) {
+          val dotPath: String = StringUriUtilities.formatUri(getPath(d, configurationRoot))
+          val out: File = new File(dotPath)
+          try {
+            if (out.exists && out.canWrite || !out.exists && out.createNewFile) {
+              handlers.add(new SaveDotHandler(out, true, true))
+            } else {
+              logger.warn("Cannot write to DOT file: " + dotPath)
+            }
+          } catch {
+            case ex: IOException =>
+              logger.warn("Cannot write to DOT file: " + dotPath, ex)
+          }
+        }
+      case _ =>
+    }
+    new DispatchHandler(handlers.toList:_*)
+  }
+
+  private def getPath(path: String, configRoot: String): String = {
+    val file: File = new File(path)
+    if (file.isAbsolute) {
+      file.getAbsolutePath
+    } else {
+      new File(configRoot, path).getAbsolutePath
+    }
+  }
+
+  def readResource(resourceStream: InputStream): Option[String] = {
+    Try(Some(Source.fromInputStream(resourceStream).getLines().mkString("\n"))).getOrElse(None)
+  }
+
+  private def rbacToWadl(rbac: Option[String]): Option[String] = {
+
+    case class Resource(path: String, methods: Set[String], roles: Set[String])
+
+    def parseLine(line: String): Option[Resource] = {
+      val values = line.split("\\s+")
+      if (values.length == 3) {
+        Some(new Resource(
+          values(0),
+          Try(values(1).split(',').toSet[String].map(_.trim)).getOrElse(Set.empty),
+          Try(values(2).split(',').toSet[String].map(_.trim)).getOrElse(Set.empty)
+        ))
+      } else {
+        logger.warn(s"Malformed RBAC Resource: $line")
+        None
+      }
+    }
+
+    val parsed = rbac match {
+      case Some(l) =>
+        Some(l.replaceAll("[\r?\n?]", "\n").split('\n').toList.flatMap(parseLine))
+      case _ =>
+        None
+    }
+
+    parsed match {
+      case Some(_) =>
+        val header = s"""<application xmlns:rax="http://docs.rackspace.com/api"
+                        |         xmlns:xsd="http://www.w3.org/2001/XMLSchema"
+                        |         xmlns="http://wadl.dev.java.net/2009/02"
+                        |    >
+                        |  <resources base="http://localhost">""".stripMargin
+        val resources = parsed.getOrElse(List.empty[Resource]).map { r =>
```

Comment:
your or else is never going to trigger, because you are in a case that says explicitly that you have a value. Maybe catch the value and use it directly instead?